packages/presentation/lib/utils/formatters.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:domain/utils/extensions/string_extension.dart';
+import 'package:flutter/services.dart';
+
+class CustomCardFormatter extends TextInputFormatter {
+  @override
+  TextEditingValue formatEditUpdate(
+    TextEditingValue oldValue,
+    TextEditingValue newValue,
+  ) {
+    final List inputChars = newValue.text.split('');
+    String result = '';
+    List newString = [];
+    if (newValue.text.length > oldValue.text.split(' ').join('').length) {
+      for (var i = 0; i < inputChars.length; i++) {
```

Comment:
I think it's easier by doing in this way:

input string 12345678 12345678 (for ex)
 |
\/
remove all spaces - 1234567812345678
 |
\/
split string by 4 symbols - [1234, 5678,1234, 5678]
 |
\/
join with spaces - 1234 5678 1234 5678